Minio/V4Authenticator.cs

```diff
@@ -414,30 +414,10 @@ private string GetPresignCanonicalRequest(IRestClient client, IRestRequest reque
             {
                 string headerName = header.Name.ToLower();
                 string headerValue = header.Value.ToString();
-#if NET46
                 if (!ignoredHeaders.Contains(headerName))
                 {
                     sortedHeaders.Add(headerName, headerValue);
                 }
-#else
-                if (headerName.Equals("host"))
-                {
-                    var host = headerValue.Split(':')[0];
-                    var port = headerValue.Split(':')[1];
-                    if (port.Equals("80") || port.Equals("443"))
-                    {
-                        sortedHeaders.Add(headerName, host);
-                    }
-                    else
-                    {
-                        sortedHeaders.Add(headerName, headerValue);
-                    }
-                }
-                else if (!ignoredHeaders.Contains(headerName))
-                {
-                    sortedHeaders.Add(headerName, headerValue);
-                }
-#endif
```

Comment:
@WolfspiritM, standard ports such as 80 and 443 are removed from the host header before generating authorization header - this is correctly handled in the .netcore specific code. 

W.r.t your presigned request for .netcore failing with signature mismatch, can you paste the actual error message from minio and the reverse proxy config?

Reply:
Why is that only handled in .NET Core then and not in .NET Framework? Both generate a different signature and with .NET Framework the connection worked while with .NET Core I got the Mismatch error.

```
Minio API responded with message=The request signature we calculated does not match the signature you provided. Check your key and signing method.'
```

I'm using traefik as reverse proxy. I now started minio with --address :80 and the following .NET core code:
``` c#
 var minioClient = new MinioClient("localhost",
                "HX4OBAN38VW39BYB5KS5",
                "ydmDD0AqRPTfqV0ZbAmsgMJl6+G2gfFkb9wKcuyc");

            var t = minioClient.BucketExistsAsync("test").GetAwaiter().GetResult();
            Console.WriteLine(t);
            Console.ReadLine();
```
and that fails aswell for .NET Core but works for .NET Framework. (just switching the Framework to net461 fixes it) so it's not a reverse proxy problem!

The NetCore version works fine for .NET Core. Only the new 2.0 version breaks and while debugging for .net framework and .net core I noticed that .net framework adds the port and .net core does not. After changing it the way described in this PR it works for both ways.

Reply:
@poornas ```<TargetFramework>netcoreapp2.1</TargetFramework>```

```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.301
 Commit:    59524873d6
```